### PR TITLE
Create Ironic format yaml

### PR DESCRIPTION
**Pull request by hardys**
This converts from the tripleo instackenv.json to the ironic native format, and updates the docs to show how we can register the masters with this file.

### CODE REVIEW

File: tripleo-quickstart-config/roles/libvirt/setup/overcloud/templates/ironic_nodes.json.j2 [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    {
+      "name": "{{ node.name|replace('_', '-') }}",
+      "driver": "ipmi",
+      "resource_class": "baremetal",
+      "driver_info": {
+        "ipmi_user": "admin",
+        "ipmi_password": "password",
+        "ipmi_addr": "192.168.122.1",
+        "ipmi_port": "{{ node.virtualbmc_port }}",
+        "deploy_kernel": "http://172.22.0.1/images/tinyipa-stable-rocky.vmlinuz",
```
Comment (dtantsur) [inline tripleo-quickstart-config/roles/libvirt/setup/overcloud/templates/ironic_nodes.json.j2]: We may parametrize the release (s/stable-rocky/.../) eventually

File: README.md [hunk 2]
```diff
@@ -46,25 +46,40 @@ e.g `sudo echo "192.168.122.235 ostest-api.test.metalkube.org" >> /etc/hosts`.
...
-`journalctl -b -f -u bootkube.service`.
+```
+openstack baremetal create ocp/master_nodes.json
+```
+
+Then to deploy you can do e.g:
+
+```
+openstack baremetal node set $NODE_UUID --instance-info image_source=https://download.cirros-cloud.net/0.3.5/cirros-0.3.5-x86_64-disk.img --instance-info image_checksum=f8ab98ff5e73ebab884d80c9dc9c7290 --instance-info root_gb=10  --property root_device='{"name": "/dev/vda"}'
+openstack baremetal node manage $NODE_UUID --wait
```
Comment (dtantsur) [inline README.md]: Strictly speaking, the `manage` action should go before populating instance_info. It does not matter, just looks less awkward to me :)

Also, may I advertise you `metalsmith` CLI as a means of testing standalone ironic? :) It's pretty neat.

File: tripleo-quickstart-config/roles/libvirt/setup/overcloud/templates/ironic_nodes.json.j2 (hunk 1 above)
Comment (hardys) [reply tripleo-quickstart-config/roles/libvirt/setup/overcloud/templates/ironic_nodes.json.j2]: Ack yeah this was a cut/paste from the readme but we can add a variable when needed

File: README.md (hunk 2 above)
Comment (hardys) [reply README.md]: Ack yeah I didn't change the order here, only fix the code block formatting - happy to switch it and/or move to metalsmith in a follow-up :)

When we've got this working I hope we'll have an init container which drives the ironic API, so we can transparently switch in whatever tool/code is preferable.

**Comment by hardys** [comment]
Got a lgtm from @derekhiggins on slack so I'll go ahead and merge this

**Pull request merged**